### ISSUE TITLE
Exclude<string, "foo"> is not working

### ISSUE THREAD

**Issue by frodo821** [microsoft/TypeScript#38254]
**TypeScript Version:**  3.8.3 or earlier

**Search Terms:** `Exclude`, `Conditional Types`

**Code**

```ts
type Foo = Exclude<string, "bar">;

// this should cause an error.
const baz: Foo = "bar";
```

**Expected behavior:**
`"bar"` is not assignable to `baz`.

**Actual behavior:**
`"bar"` is assignable to `baz` without any errors.

**Playground Link:** https://www.typescriptlang.org/play/#code/FAFwngDgpgBAYgewTAvDAogDwMYBsCuAJlADwDOIATgJYB2A5gDQwBEARgIaUsB8A3MGDYEtCjE4AvAFzwkqVp24ChIsglxQAdLgT0AFJICUAoA

**Comment by stormpat** [comment]
Not the most elegant, but here is a work-around.

https://www.typescriptlang.org/play/?ssl=4&ssc=1&pln=5&pc=1#code/C4TwDgpgBAQghgOwRAJlAvFARAIzgJywG4AoUSKAMQHtqMoBRADwGMAbAVxQgB4BnYPgCWCAOYAaWImQoAfKRItqCAVDwAvKAC4qtergLEoAemNU4Qtn110h1uFAHCxJEgDMOCFsCHKo1AGseABUoCCZgCAQUaycRUVkACgA3HVCAMihE0PDI6Ot4JFQoAH5sGABBADkqhgARLB0AbwBfAEo2qCaSKCh8CGAOfAQoZNIW1yUVYDU4dQB9Nws2ekDEg0JO03NLPkVlVQ15wNWA9YRqebxNkzMAYQALCBYAviA

**Comment by IllusionMH** [comment]
This is limitation of `Exclude`. Basically it works when you "filter" unions, but `string` is not a union.

What you are lookin is Negated types: #29317 and comment about your use case https://github.com/microsoft/TypeScript/pull/29317#issuecomment-452987604

**Comment by frodo821** [comment]
Thank you for comments!
I guess it means that, in the type system of typescript, string type is not a 'representative' or 'set' of any valid strings.
I thought string means the set of any of it including `'foo'`, `'bar'` or any others of string.

**Comment by DanielRosenwasser** [comment]
Yeah, unfortunately we're not wired up to "cut holes" out of infinitely large domains  of types, and it appears that adding that assumption to the type system would make it incredibly painful to use.

Regarding

> https://www.typescriptlang.org/play/?ssl=4&ssc=1&pln=5&pc=1#code/C4TwDgpgBAQghgOwRAJlAvFARAIzgJywG4AoUSKAMQHtqMoBRADwGMAbAVxQgB4BnYPgCWCAOYAaWImQoAfKRItqCAVDwAvKAC4qtergLEoAemNU4Qtn110h1uFAHCxJEgDMOCFsCHKo1AGseABUoCCZgCAQUaycRUVkACgA3HVCAMihE0PDI6Ot4JFQoAH5sGABBADkqhgARLB0AbwBfAEo2qCaSKCh8CGAOfAQoZNIW1yUVYDU4dQB9Nws2ekDEg0JO03NLPkVlVQ15wNWA9YRqebxNkzMAYQALCBYAviA

you might be able to get away with using `never` in place of `BANNED`
